src/main/java/uk/gov/ch/developer/docs/controller/developer/SignOutController.java

```diff
@@ -1,30 +1,60 @@
 package uk.gov.ch.developer.docs.controller.developer;
 
 import java.io.IOException;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import uk.gov.ch.developer.docs.DocsWebApplication;
-import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.ch.oauth.IIdentityProvider;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
+import uk.gov.companieshouse.session.Session;
+import uk.gov.companieshouse.session.SessionKeys;
+import uk.gov.companieshouse.session.handler.SessionHandler;
+import uk.gov.companieshouse.session.store.Store;
+import uk.gov.companieshouse.session.store.StoreImpl;
 
 @Controller
 public class SignOutController {
-    private static final Logger LOGGER = LoggerFactory.getLogger(DocsWebApplication.APPLICATION_NAME_SPACE);
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DocsWebApplication.APPLICATION_NAME_SPACE);
 
-    private final String signOutUrl;
-
-    public SignOutController(final EnvironmentReader reader) {
-        this.signOutUrl = reader.getMandatoryString("CHS_SIGNOUT_URL");
-    }
+    @Autowired
+    private IIdentityProvider identityProviders;
 
     @GetMapping("${signout.url}")
-    public void doSignOut(final HttpServletResponse httpServletResponse) throws IOException {
+    public void doSignOut(final HttpServletResponse httpServletResponse,
+            final HttpServletRequest httpServletRequest) throws IOException {
+
+        LOGGER.debug("Processing sign out");
+
+        final Session chSession = (Session) httpServletRequest
+                .getAttribute(SessionHandler.CHS_SESSION_REQUEST_ATT_KEY);
+
+        if (chSession.getSignInInfo().isSignedIn()) {
+
+            Map<String, Object> sessionData = chSession.getData();
+            Map<String, Object> signInInfo =
+                    (Map<String, Object>) sessionData.get(SessionKeys.SIGN_IN_INFO.getKey());
+
+            signInInfo.replace(SessionKeys.SIGNED_IN.getKey(), 1, 0);
+
```

Comment:
There seem to be a lot of unnecessary blank lines, could this be tidied up, please?